src/compiler/core.ts

```diff
@@ -2334,15 +2334,34 @@ namespace ts {
             return currentAssertionLevel >= level;
         }
 
-        export function assert(expression: boolean, message?: string, verboseDebugInfo?: () => string, stackCrawlMark?: Function): void {
+        export function assert(expression: boolean, message?: string, verboseDebugInfo?: string | (() => string), stackCrawlMark?: Function): void {

```

Comment:
This avoids needing a closure for the second argument (if it is not an interpolation).